### PR TITLE
fix(ci-visibility): do not enable CIVisibility service in agentless mode if DD_API_KEY is not set

### PR DESCRIPTION
**Pull request by romainkomorndatadog**
Fixes the behavior of the CIVisibility service to simply not enable (instead of raising an exception) if the `DD_API_KEY` environment variable is not set when the `DD_CIVISIBILITY_AGENTLESS_ENABLE` one is set and truthy.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Comment by romainkomorndatadog** [comment]
I'm split between this implementation, or raising an `EnvironmentError` and catching when the `pytest` plugin calls `CIVisibility.enable()`.

Happy to go the latter route if reviewers feel strongly (or even not strongly, really).

**Comment by gnufede** [comment]
> I'm split between this implementation, or raising an `EnvironmentError` and catching when the `pytest` plugin calls `CIVisibility.enable()`.
> 
> Happy to go the latter route if reviewers feel strongly (or even not strongly, really).

I like this approach, but does it make sense to keep these lines?
https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/internal/ci_visibility/recorder.py#L129

**Comment by romainkomorndatadog** [comment]
> I like this approach, but does it make sense to keep these lines? https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/internal/ci_visibility/recorder.py#L129

I kept them because I was paranoid about someone calling doing something like `mything = CIVisibility(...)` instead of setting the `_instance` via the `CIVisibility.enable()` classmethod, but in our current flow, at least, they're theoretically unreachable.

**Review by majorgreys — APPROVED** [review]

File: ddtrace/internal/ci_visibility/recorder.py [hunk 1]
```diff
@@ -368,11 +368,21 @@ def _should_skip_path(self, path, name, test_skipping_mode=None):
     @classmethod
     def enable(cls, tracer=None, config=None, service=None):
         # type: (Optional[Tracer], Optional[Any], Optional[str]) -> None
+        log.debug("Enabling %s", cls.__name__)
+
+        if ddconfig._ci_visibility_agentless_enabled:
+            if not os.getenv("_CI_DD_API_KEY", os.getenv("DD_API_KEY")):
```
[the remark this answers is not in the document]
Comment (romainkomorndatadog) [reply ddtrace/internal/ci_visibility/recorder.py:374]: It's the logic from https://github.com/DataDog/dd-trace-py/pull/6583/files#diff-3c04fbe5dfdadd108e9a074d21c671958693d6e77758c3b0144df89491286493R107 .

I think in practice, `_CI_DD_API_KEY` and `DD_API_KEY` are unlikely to both be set anyway, bit if the `_CI` one is set, it's the one we want to use.

**Review by P403n1x87 — APPROVED** [review]

**Review by gnufede — APPROVED** [review]

**Review by emmettbutler — APPROVED** [review]

**Pull request merged**